src/libstore/build.cc

```diff
@@ -1452,32 +1542,35 @@ void DerivationGoal::tryToBuild()
        omitted, but that would be less efficient.)  Note that since we
        now hold the locks on the output paths, no other process can
        build this derivation, so no further checks are necessary. */
-    validPaths = checkPathValidity(true, buildMode == bmRepair);
-    if (buildMode != bmCheck && validPaths.size() == drv->outputs.size()) {
+    bool allValid = true;
+    for (auto & [_, status] : initialOutputs) {
```

Comment:
Good point. I was thinking we already had the information but didn't consider that it could change. We just acquired some locks above this, so this a good time to check (again).